contracts/Xorshift128plus.cdc

```diff
@@ -26,6 +26,7 @@ access(all) contract Xorshift128plus {
         init(sourceOfRandomness: [UInt8], salt: [UInt8]) {
             pre {
                 sourceOfRandomness.length >= 16: "At least 16 bytes of entropy should be used"
+                Xorshift128plus.isNotZeroArray(sourceOfRandomness): "Source of randomness cannot be all zeros"
```

Comment:
This is not the array that needs to be checked. We should check the input to the xorshift128+ itself (16 bytes). That can be checked either:
- by checking the `seed` array below
- Or the values `segment0` and `segment1` (this would be easier - it doesn't require `isNotZeroArray`)  

I didn't implement it myself because I didn't know how to add a check outside the `pre {}` section.

Reply:
Ah alright, I see now. Sorry for the misunderstanding. Should be a quick change.

Reply:
Updated and removed initialization test case since we don't have a pre-image that would generate a failing case.